backend/app/database/core.py

```
@@ -0,0 +1,57 @@
+    """
+    if not async_session_maker:
+        raise RuntimeError("Database engine is not initialized. check DATABASE_URL.")
+    
+    async with async_session_maker() as session:
+        try:
+            yield session
+            # automatic commit/rollback is often handled by caller or service layer logic
+            # but standard practice for read-heavy apps is just to close.
+            # If explicit commit is needed, service layer should do it.
+        except Exception as e:
+            logger.error(f"Database session error: {e}")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)